OpenSim/Simulation/Model/Station.h

```diff
@@ -37,38 +36,56 @@ class Body;
 //=============================================================================
 //=============================================================================
 /**
- * A class implementing a Station. A Station is a point fixed to and defined with 
- * respect to a reference frame. The reference frame can be a Body (fixed 
- * to origin of a Body), a FixedFrame (affixed to another PhysicalFrame) or any 
- * other PhysicalFrame. The main functionality provided by Station is to find its 
- * location in any Frame.
+ * A Station is a Point fixed to and located on a PhysicalFrame, which can be a 
+ * Body (w.r.t. the Body origin), Ground or any PhysicalOffsetFrame.
+ * A Station is an analogous physical Point to a PhyscialFrame where joints,
+ * constraints and forces can be attached and/or applied.
  *
- * @author Ayman Habib
- * @version 1.0
+ * @author Ayman Habib, Ajay Seth
  */
-class OSIMSIMULATION_API Station : public ModelComponent {
-OpenSim_DECLARE_CONCRETE_OBJECT(Station, ModelComponent);
+class OSIMSIMULATION_API Station : public Point {
+OpenSim_DECLARE_CONCRETE_OBJECT(Station, Point);
 public:
     //==============================================================================
     // PROPERTIES
     //==============================================================================
+    /** @name Property declarations
+    These are the serializable properties associated with a Station. **/
+    /**@{**/
     OpenSim_DECLARE_PROPERTY(location, SimTK::Vec3,
-        "The location (Vec3) of the station in a reference frame. "
-        "Frame is specified as Connector.");
+        "The location of the station in its reference frame.");
+    /**@}**/
 
 public:
     //--------------------------------------------------------------------------
     // CONSTRUCTION
     //--------------------------------------------------------------------------
+    /** Default constructor */
     Station();
+
+    /** Convenience constructor
+    @param[in] frame     PhysicalFrame in which the Station is located.
+    @param[in] location  Vec3 location of the station in its PhysicalFrame */
+    Station(const PhysicalFrame& frame, const SimTK::Vec3& location);
+
     virtual ~Station();
     /** getter of Reference Frame off which the Station is defined */
-    const OpenSim::PhysicalFrame& getReferenceFrame() const;
+    const PhysicalFrame& getReferenceFrame() const;
```

Comment:
I personally think we could have a better name for this method; I think `getFrame()` is fine (and the corresponding connector would also be `frame`). The main issue is that we don't use the term `ReferenceFrame` elsewhere in the code, so users may think it is something different from a Frame. I remember discussing this with @aymanhab and he had a reason for why he chose ReferenceFrame. @aymanhab, do you remember?